Added readme for pruneOFA examples on SQuAD and GLUE

XinyuYe-Intel opened the pull request:
# What does this PR do?

Add readme for pruneOFA examples on SQuAD and GLUE

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum-intel/pr_72). All of your documentation changes will be reflected on that endpoint.

hshen14 reviewed (approved):
LGTM

hshen14 commented:
@echarlaix This PR is to add some examples to reproduce the results published in NeurIPS'21 ENLSP paper "Prune Once for All: Sparse Pre-Trained Language Models" (link: https://arxiv.org/abs/2111.05754). Please review. Thanks.

echarlaix commented on examples/neural_compressor/text-classification/README.md:
```diff
@@ -73,6 +73,61 @@ python run_glue.py \
 
+### Prune Once For All
+
+The following example demonstrate the steps of reproducing Prune Once For All examples results on GLUE tasks.
```
```suggestion
The following example demonstrate the steps of reproducing Prune Once For All examples results on the sst-2 task.
```

echarlaix commented on examples/neural_compressor/text-classification/README.md:
```diff
@@ -73,6 +73,61 @@ python run_glue.py \
+
+The following example demonstrate the steps of reproducing Prune Once For All examples results on GLUE tasks.
+<br>
+This examples will take the pre-trained sparse language model and fine tune it on the downstream task. This fine tune pipeline is two staged. For stage 1, the pattern lock pruning and the distillation are applied to fine tune the pre-trained sparse language model. In stage 2, the pattern lock pruning, distillation and quantization aware training are performed simultaneously on the fine tuned mode ...
```
```suggestion
This examples will take a pre-trained DistilBERT with a sparsity of 90% and fine-tune it on the downstream task. This fine-tuning process will be decomposed into two steps. During step 1, distillation as well as pattern lock pruning are both applied during fine-tuning. During step 2, quantization aware training will be additionaly applied, to obtain a fine-tuned quantized model with the same sparsity as the pre-trained model.
```

echarlaix commented on examples/neural_compressor/question-answering/README.md:
```diff
@@ -78,6 +78,63 @@ python run_qa.py \
 
+## Prune Once For All
+
+The following example demonstrate the steps of reproducing Prune Once For All examples results on SQuAD dataset.
```
```suggestion
The following example demonstrate the steps of reproducing Prune Once For All examples results on SQuAD1.0 dataset.
```

echarlaix commented on examples/neural_compressor/question-answering/README.md:
```diff
@@ -78,6 +78,63 @@ python run_qa.py \
+
+The following example demonstrate the steps of reproducing Prune Once For All examples results on SQuAD dataset.
+<br>
+This examples will take the pre-trained sparse language model and fine tune it on the downstream task. This fine tune pipeline is two staged. For stage 1, the pattern lock pruning and the distillation are applied to fine tune the pre-trained sparse language model. In stage 2, the pattern lock pruning, distillation and quantization aware training are performed simultaneously on the fine tuned mode ...
```
```suggestion
This examples will take a pre-trained DistilBERT with a sparsity of 90% and fine-tune it on the downstream task. This fine-tuning process will be decomposed into two steps. During step 1, distillation as well as pattern lock pruning are both applied during fine-tuning. During step 2, quantization aware training will be additionaly applied, to obtain a fine-tuned quantized model with the same sparsity as the pre-trained model.
```

echarlaix reviewed (commented):
Great idea to add prunce once for all it in the README. I added some minor comments concerning the two steps explanation

XinyuYe-Intel commented:
> Great idea to add prunce once for all it in the README. I added some minor comments concerning the two steps explanation

@echarlaix Thanks for your suggestions! I've modified the readmes accordingly.

echarlaix reviewed (approved):
LGTM, thanks a lot @XinyuYe-Intel !
